core/src/main/java/org/elasticsearch/search/aggregations/metrics/avg/AvgAggregationBuilder.java

```diff
@@ -36,18 +37,19 @@
 import java.io.IOException;
 
 public class AvgAggregationBuilder extends ValuesSourceAggregationBuilder.LeafOnly<ValuesSource.Numeric, AvgAggregationBuilder> {
-    public static final String NAME = InternalAvg.TYPE.name();
+    public static final String NAME = "avg";
+    private final static Type TYPE = new Type("avg");
```

Comment:
Should `TYPE` here reference the `NAME` constant? so `private final static Type TYPE = new Type(NAME);`